src/sync_connection_wrapper.rs

```diff
@@ -206,9 +206,11 @@ impl<C> SyncConnectionWrapper<C> {
     {
         let inner = self.inner.clone();
         tokio::task::spawn_blocking(move || {
-            let mut inner = inner
-                .lock()
-                .expect("Mutex is poisoned, a thread must have panicked holding it.");
+            let mut inner = inner.lock().unwrap_or_else(|poison| {
+                // try to be resilient by providing the guard
+                inner.clear_poison();
+                poison.into_inner()
+            });
```

Comment:
I just wonder whether it would be simpler to just always call `.clear_poison()` before lock here?